pkg/operator/ceph/cluster/mgr/drain.go

```
@@ -0,0 +1,101 @@
+	usePDBV1Beta1, err := k8sutil.UsePDBV1Beta1Version(c.context.Clientset)
+	if err != nil {
+		return controllerutil.OperationResultNone, errors.Wrap(err, "failed to fetch pdb version")
+	}
+	objectMeta := metav1.ObjectMeta{
+		Name:      mgrPDBName,
+		Namespace: c.clusterInfo.Namespace,
+	}
+	selector := &metav1.LabelSelector{
+		MatchLabels: map[string]string{k8sutil.AppAttr: AppName},
+	}
+	if usePDBV1Beta1 {
+		pdb := &policyv1beta1.PodDisruptionBudget{
+			ObjectMeta: objectMeta,
+		}
+		mutateFunc := func() error {
+			pdb.Spec = policyv1beta1.PodDisruptionBudgetSpec{
+				Selector:       selector,
+				MaxUnavailable: &intstr.IntOrString{IntVal: maxUnavailable},
+			}
+			return nil
+		}
+		return controllerutil.CreateOrUpdate(context.TODO(), c.context.Client, pdb, mutateFunc)
+	}
+	pdb := &policyv1.PodDisruptionBudget{
+		ObjectMeta: objectMeta,
+	}
+	mutateFunc := func() error {
+		pdb.Spec = policyv1.PodDisruptionBudgetSpec{
+			Selector:       selector,
+			MaxUnavailable: &intstr.IntOrString{IntVal: maxUnavailable},
+		}
+		return nil
+	}
+	return controllerutil.CreateOrUpdate(context.TODO(), c.context.Client, pdb, mutateFunc)
+}
+
+func (c *Cluster) deleteMgrPDB() {
+	pdbRequest := types.NamespacedName{Name: mgrPDBName, Namespace: c.clusterInfo.Namespace}
+	usePDBV1Beta1, err := k8sutil.UsePDBV1Beta1Version(c.context.Clientset)
+	if err == nil {
+		if usePDBV1Beta1 {
+			mgrPDB := &policyv1beta1.PodDisruptionBudget{}
+			err := c.context.Client.Get(context.TODO(), pdbRequest, mgrPDB)
+			if err == nil {
+				logger.Infof("deleting the mgr pdb %q with maxUnavailable=1", mgrPDBName)
+				_ = c.context.Client.Delete(context.TODO(), mgrPDB)
+			}
+		}
+		mgrPDB := &policyv1.PodDisruptionBudget{}
+		err = c.context.Client.Get(context.TODO(), pdbRequest, mgrPDB)
+		if err == nil {
+			logger.Infof("deleting the mgr pdb %q with maxUnavailable=1", mgrPDBName)
+			_ = c.context.Client.Delete(context.TODO(), mgrPDB)
```

instead of ignore the error, how about logging it?
```suggestion
			err = c.context.Client.Delete(context.TODO(), mgrPDB)
			if err != nil && !errors.IsNotFound(err) {
				logger.Errorf(...)
			}
```

---

Done